pkg/encoding/encoding.go

```diff
@@ -17,6 +17,9 @@ func Decode(data []byte) (*spec.App, error) {
 		return nil, errors.Wrap(err, "could not unmarshal into internal struct")
 	}
 	log.Debugf("object unmrashalled: %#v\n", app)
+	if err := sanifyApp(&app); err != nil {
```

Comment:
@kadel I have refactored it to `fixApp`, etc. The idea is to fix whatever is wrong, missing, or additional in `spec.App`